java/ddlogapi/DDlogRecord.java

```diff
@@ -380,12 +397,20 @@ public String getStructName() {
         return DDlogAPI.ddlog_get_constructor(this.handle);
     }
 
-    public DDlogRecord getStructField(int index) {
+    public DDlogRecord getStructField(int index) throws DDlogException {
         if (!this.isStruct())
             throw new RuntimeException("Value is not a struct");
         return fromSharedHandle(DDlogAPI.ddlog_get_struct_field(this.handle, index));
     }
 
+    public DDlogRecord getStructFieldNoEx(int index) {
```

Comment:
Better to call this getStructFieldUnchecked.

Reply:
It's checked, but not a checked exception.